tests/test_urlutils.py

```diff
@@ -268,6 +268,23 @@ def test_rel_navigate():
     return
 
 
+@pytest.mark.parametrize(
+    ('expected',            'base',             'a',    'b'), [
+    ('https://host/b',      'https://host',     'a',    '/b'),
+    ('https://host/a/b',    'https://host',     'a',    'b'),
```

Comment:
Using
```
boltons.urlutils.URL(boltons.urlutils.URL('https://host').navigate('a')).navigate('b')
```
the expected result should be `https://host/b`

Reply:
I haven't looked at this in a long time, so pardon me if I'm completely backwards and no longer clear on what I thought behavior should be.

When you say, "the expected result should be", where are you basing that "should be" on?  The scant documentation says, "Useful for navigating those relative links with ease." I consider this analogous to using `cd` on a Unix system; all references are relative unless the destination starts with `/`.

Actually, the vexing nature of the way things work now is more apparent if you have multiple path segments:
```
>>> boltons.urlutils.URL('https://host/a/b/c').navigate('d')
URL('https://host/a/b/d')
>>> boltons.urlutils.URL('https://host/a/b/c').navigate('../d')
URL('https://host/a/d')
>>> boltons.urlutils.URL('https://host/a/b/c/').navigate('../d')
URL('https://host/a/b/d')
>>> boltons.urlutils.URL('https://host/a/b/c').navigate('#d')
URL('https://host/a/b/c#d')
```

What I find most surprising is that, without a trailing slash on the base, `navigate` with a non-fragment argument _assumes_ the last part of the path is (erm, not sure about the right terminology) a "leaf" instead of a "stem" (or "file" instead of a "directory"?).

 To make a relative link from an URL without a trialing slash with the behavior you're describing, you have to: 1. `to_text()` 2. conditionally append a '/' (`normalize()` will not make `//` -> `/`) 3. recreate URL. Because the `URL` object is immutable, it's much easier to change the string argument to enable "upwards" navigation by pre-pending '../'.  This is what I would find less surprising:
```
>>> boltons.urlutils.URL('https://host/a/b/c').navigate('d')
URL('https://host/a/b/c/d')
>>> boltons.urlutils.URL('https://host/a/b/c').navigate('../d')
URL('https://host/a/b/d')
>>> boltons.urlutils.URL('https://host/a/b/c/').navigate('../d')
URL('https://host/a/b/d')
```

Unfortunately, I think this ship has sailed; changing this behavior would be massively backwards-incompatible.